shapely/tests/test_constructive.py

```diff
@@ -901,74 +901,54 @@ def test_oriented_envelope_all_types(geometry):
+    ),
+    (
+        LineString([(1, 1), (5, 1), (10, 10)]),
+        Polygon([(1, 1), (3, -1), (12, 8), (10, 10), (1, 1)]),
+    ),
+    (
+        Polygon([(1, 1), (15, 1), (5, 10), (1, 1)]),
+        Polygon([(15, 1), (15, 10), (1, 10), (1, 1), (15, 1)]),
+    ),
+    (
+        LineString([(1, 1), (10, 1)]),
+        LineString([(1, 1), (10, 1)]),
+    ),
+    (
+        Point(2, 2),
+        Point(2, 2),
+    ),
+    (
+        GeometryCollection(),
+        Polygon(),
+    ),
+]
+
+
 @pytest.mark.skipif(shapely.geos_version < (3, 6, 0), reason="GEOS < 3.6")
-@pytest.mark.parametrize(
-    "geometry, expected",
-    [
-        (
-            MultiPoint([(0, 0), (10, 0), (10, 10)]),
-            Polygon([(0, 0), (5, -5), (15, 5), (10, 10), (0, 0)]),
-        ),
-        (
-            LineString([(1, 1), (5, 1), (10, 10)]),
-            Polygon([(1, 1), (3, -1), (12, 8), (10, 10), (1, 1)]),
-        ),
-        (
-            Polygon([(1, 1), (15, 1), (5, 10), (1, 1)]),
-            Polygon([(15, 1), (15, 10), (1, 10), (1, 1), (15, 1)]),
-        ),
-        (
-            LineString([(1, 1), (10, 1)]),
-            LineString([(1, 1), (10, 1)]),
-        ),
-        (
-            Point(2, 2),
-            Point(2, 2),
-        ),
-        (
-            GeometryCollection(),
-            Polygon(),
-        ),
-    ],
-)
-def test_oriented_envelope(geometry, expected):
-    actual = shapely.oriented_envelope(geometry)
-    assert shapely.equals(actual, expected).all()
+@pytest.mark.parametrize("geometry, expected", oriented_envelop_cases)
+@pytest.mark.parametrize("method", [None, ..., "min_area", "unknown"])
+def test_oriented_envelope(geometry, expected, method):
+    if method == "unknown":
+        with pytest.raises(ValueError, match=f"Unknown method {method}"):
+            shapely.oriented_envelope(geometry, method)
+    else:
+        actual = shapely.oriented_envelope(geometry, method).normalize()
+        assert_geometries_equal(actual, expected, normalize=True)
 
 
 @pytest.mark.skipif(shapely.geos_version < (3, 6, 0), reason="GEOS < 3.6")
-@pytest.mark.parametrize(
-    "geometry, expected",
-    [
-        (
-            MultiPoint([(0, 0), (10, 0), (10, 10)]),
-            Polygon([(0, 0), (5, -5), (15, 5), (10, 10), (0, 0)]),
-        ),
-        (
-            LineString([(1, 1), (5, 1), (10, 10)]),
-            Polygon([(1, 1), (3, -1), (12, 8), (10, 10), (1, 1)]),
-        ),
-        (
-            Polygon([(1, 1), (15, 1), (5, 10), (1, 1)]),
-            Polygon([(15, 1), (15, 10), (1, 10), (1, 1), (15, 1)]),
-        ),
-        (
-            LineString([(1, 1), (10, 1)]),
-            LineString([(1, 1), (10, 1)]),
-        ),
-        (
-            Point(2, 2),
-            Point(2, 2),
-        ),
-        (
-            GeometryCollection(),
-            Polygon(),
-        ),
-    ],
+@pytest.mark.skipif(
+    shapely.minimum_rotated_rectangle == shapely.oriented_envelope,
```

Comment:
Currently yes, maybe in some future version some decides they are not two names for the same function.
Otherwise do you want to test both functions or just one ?

Reply:
Yes, we want to test both (we want to ensure that both give the same result)